projects/saturn/src/main/java/io/fairspace/saturn/webdav/DavFactory.java

```diff
@@ -126,16 +126,42 @@ org.apache.jena.rdf.model.Resource currentUserResource() {
         return rootSubject.getModel().createResource(getUserURI().getURI());
     }
 
-    public boolean isFileSystemResource(org.apache.jena.rdf.model.Resource resource) {
-        return resource.getURI().startsWith(rootSubject.getURI());
-    }
-
     public void linkEntityToSubject(org.apache.jena.rdf.model.Resource subject) throws BadRequestException {
         var linkedEntity = getNewLinkedEntity(subject);
         subject.addProperty(FS.linkedEntity, linkedEntity);
         subject.addProperty(FS.linkedEntityType, linkedEntity.getPropertyResourceValue(RDF.type));
     }
 
+    public String getEntityTypeFromRequest() throws BadRequestException {
```

Comment:
I don't think this should do all kinds of validations that are already done (exact same ones) at later stages. 
Maybe let's discuss this tomorrow.